src/core/Clock.js

```diff
@@ -2,6 +2,10 @@
  * @author alteredq / http://alteredqualia.com/
  */
 
+/**
+ * @class
```

Comment:
`@class` tells TS that this is a constructor function, which make it only callable with `new`. Additionally it make `this` properly typed within the body.